### PR TITLE
#16974 check for null, if no schemas

### LINKED ISSUE #16974: Creating ER Diagram for Access (*.accdb) fails with NullPointerException
#### System information: 
- Operating system: Windows 10 Enterprise, Build 19044.1766
- DBeaver version:  22.1.1.202206261508

#### Connection specification:
- Microsoft Access Database
- Driver name: MS Access (UCanAccess)
- Do you use tunnels or proxies?: none

#### Describe the problem you're observing:
As soon as I hit the *Finish* button in the *Create new diagram* dialog I am getting a NullPointerException from Plug-In org.jkiss.dbeaver.model.

#### Steps to reproduce, if exist:
- Create connection to a local Access database (\*.accdb)
- Open *Create new diagram* dialog
- Check the Access database and provide a name for the diagram
- Click the *Finish* button of the dialog

#### Include any warning/errors/backtraces from the logs
eclipse.buildId=unknown
java.version=11.0.12
java.vendor=Eclipse Foundation
BootLoader constants: OS=win32, ARCH=x86_64, WS=win32, NL=en
Command-line arguments:  -os win32 -ws win32 -arch x86_64

org.jkiss.dbeaver.model
Error
Wed Jul 06 09:30:34 CEST 2022
NullPointerException

java.lang.NullPointerException
	at org.jkiss.dbeaver.ext.generic.model.GenericDataSource.cacheStructure(GenericDataSource.java:718)
	at org.jkiss.dbeaver.erd.model.DiagramObjectCollector.collectTables(DiagramObjectCollector.java:113)
	at org.jkiss.dbeaver.erd.model.DiagramObjectCollector.collectTables(DiagramObjectCollector.java:97)
	at org.jkiss.dbeaver.erd.model.DiagramObjectCollector.collectTables(DiagramObjectCollector.java:60)
	at org.jkiss.dbeaver.erd.ui.navigator.DiagramCreateWizard$DiagramCreator.run(DiagramCreateWizard.java:163)
	at org.jkiss.dbeaver.ui.UIUtils.lambda$13(UIUtils.java:1623)
	at org.eclipse.jface.operation.ModalContext$ModalContextThread.run(ModalContext.java:122)

### PR DESCRIPTION
**Pull request by Destrolaric**
Closes #16974

### CODE REVIEW

File: plugins/org.jkiss.dbeaver.ext.generic/src/org/jkiss/dbeaver/ext/generic/model/GenericDataSource.java [hunk 1]
```diff
@@ -715,7 +715,7 @@ public Class<? extends DBSObject> getPrimaryChildType(@Nullable DBRProgressMonit
     public void cacheStructure(@NotNull DBRProgressMonitor monitor, int scope) throws DBException {
         if (!CommonUtils.isEmpty(catalogs)) {
             for (GenericCatalog catalog : catalogs) catalog.cacheStructure(monitor, scope);
-        } else if (!schemas.isEmpty()) {
+        } else if (schemas != null && !schemas.isEmpty()) {
```
Comment (E1izabeth) [inline plugins/org.jkiss.dbeaver.ext.generic/src/org/jkiss/dbeaver/ext/generic/model/GenericDataSource.java:718]: But isEmpty argument is nullable and checks for null
```
    public static boolean isEmpty(@Nullable Collection<?> value) {
        return value == null || value.isEmpty();
    }
```

Comment (E1izabeth) [reply plugins/org.jkiss.dbeaver.ext.generic/src/org/jkiss/dbeaver/ext/generic/model/GenericDataSource.java:718]: Maybe it's better to rethink isEmpty method of the AbstractObjectCache?

**Review by E1izabeth — APPROVED** [review]

File: plugins/org.jkiss.dbeaver.ext.generic/src/org/jkiss/dbeaver/ext/generic/model/GenericDataSource.java (hunk 1 above)
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.ext.generic/src/org/jkiss/dbeaver/ext/generic/model/GenericDataSource.java:718]:
```suggestion
        } else if (!CommonUtils.isEmpty(schemas)) {
```

**Review by ShadelessFox — CHANGES_REQUESTED** [review]

**Review by ShadelessFox — APPROVED** [review]

**Review by ShadelessFox — APPROVED** [review]

**Comment by Matvey16** [comment]
Verified

**Pull request merged**